RHEL 8 build broken for GCE

## Environment

- **Make target:** `make gce-rhel-8`
/kind bug

Running the `pull-image-builder-gcp-all` target in CI failed on rhel-8. See #1515 for an example:

```
...
    googlecompute.rhel-8: TASK [providers : ansible.builtin.include_tasks] *******************************
    googlecompute.rhel-8: included: /home/prow/go/src/sigs.k8s.io/image-builder/images/capi/ansible/roles/providers/tasks/googlecompute.yml for default
    googlecompute.rhel-8:
    googlecompute.rhel-8: TASK [providers : Download gcloud SDK] *****************************************
    googlecompute.rhel-8: changed: [default]
    googlecompute.rhel-8:
    googlecompute.rhel-8: TASK [providers : Execute install-gcloud.sh] ***********************************
    googlecompute.rhel-8: fatal: [default]: FAILED! => {"changed": true, "cmd": "bash -o errexit -o pipefail /tmp/install-gcloud.sh --disable-prompts --install-dir=/", "delta": "0:00:06.086712", "end": "2024-07-18 21:05:22.335836", "msg": "non-zero return code", "rc": 1, "start": "2024-07-18 21:05:16.249124", "stderr": "#=#=#                                                                         \r\r######################################################################## 100.0%\nwhich curl\ncurl -# -f https://dl.google.com/dl/cloudsdk/channels/rapid/google-cloud-sdk.tar.gz\n#=#=#                                                                         \r\r                                                                           0.3%\r######################################################################## 100.0%\nmkdir -p /\ntar -C / -zxvf /tmp/tmp.x2lbfShoyl/google-cloud-sdk.tar.gz\n//google-cloud-sdk/install.sh\nWARNING:  Python 3.6.x is no longer officially supported by the Google Cloud CLI\nand may not function correctly. Please use Python version 3.8 and up.\n\nIf you have a compatible Python interpreter installed, you can use it by setting\nthe CLOUDSDK_PYTHON environment variable to point to it.\n\nTraceback (most recent call last):\n  File \"//google-cloud-sdk/bin/bootstrapping/install.py\", line 12, in \n    import bootstrapping\n  File \"/google-cloud-sdk/bin/bootstrapping/bootstrapping.py\", line 50, in \n    from googlecloudsdk.core.credentials import store as c_store\n  File \"/google-cloud-sdk/lib/googlecloudsdk/core/credentials/store.py\", line 34, in \n    from googlecloudsdk.api_lib.auth import external_account as auth_external_account\n  File \"/google-cloud-sdk/lib/googlecloudsdk/api_lib/auth/external_account.py\", line 24, in \n    from googlecloudsdk.core.credentials import creds as c_creds\n  File \"/google-cloud-sdk/lib/googlecloudsdk/core/credentials/creds.py\", line 36, in \n    from google.auth import external_account as google_auth_external_account\n  File \"/google-cloud-sdk/lib/third_party/google/auth/external_account.py\", line 32, in \n    from dataclasses import dataclass\nModuleNotFoundError: No module named 'dataclasses'", "stderr_lines": ["#=#=#  
...
    googlecompute.rhel-8:
    googlecompute.rhel-8: PLAY RECAP *********************************************************************
    googlecompute.rhel-8: default                    : ok=18   changed=13   unreachable=0    failed=1    skipped=244  rescued=0    ignored=0
    googlecompute.rhel-8:
==> googlecompute.rhel-8: Provisioning step had errors: Running the cleanup provisioner, if present...
==> googlecompute.rhel-8: Deleting instance...
heartbeat sent for resource "k8s-infra-e2e-boskos-081"
    googlecompute.rhel-8: Instance has been deleted!
==> googlecompute.rhel-8: Deleting disk...
    googlecompute.rhel-8: Disk has been deleted!
Build 'googlecompute.rhel-8' errored after 7 minutes 26 seconds: Error executing Ansible: Non-zero exit status: exit status 2
==> Wait completed after 7 minutes 26 seconds
==> Some builds didn't complete successfully and had errors:
--> googlecompute.rhel-8: Error executing Ansible: Non-zero exit status: exit status 2
==> Builds finished but no artifacts were created.
make: *** [Makefile:480: build-gce-rhel-8] Error 1
```

--- a/images/capi/ansible/roles/providers/tasks/googlecompute.yml
+++ b/images/capi/ansible/roles/providers/tasks/googlecompute.yml
@@ -12,32 +12,50 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 ---
-- name: Download gcloud SDK
-  ansible.builtin.get_url:
-    url: https://sdk.cloud.google.com/
-    dest: /tmp/install-gcloud.sh
-    mode: "0700"
+- name: Install gcloud SDK
+  when: ansible_os_family != "RedHat"
+  block:
+    - name: Download gcloud SDK
+      ansible.builtin.get_url:
+        url: https://sdk.cloud.google.com/
+        dest: /tmp/install-gcloud.sh
+        mode: "0700"
+    - name: Execute install-gcloud.sh
+      ansible.builtin.shell: bash -o errexit -o pipefail /tmp/install-gcloud.sh --disable-prompts --install-dir=/
+    - name: Remove install-gcloud.sh
+      ansible.builtin.file:
+        path: /tmp/install-gcloud.sh
+        state: absent
+    - name: Find all files in /google-cloud-sdk/bin/
+      ansible.builtin.find:
+        paths: /google-cloud-sdk/bin/
+      register: find
+    - name: Create symlinks to /bin
+      become: true
+      ansible.builtin.file:
+        src: "{{ item.path }}"
+        path: /bin/{{ item.path | basename }}
+        state: link
+      with_items: "{{ find.files }}"
 
-- name: Execute install-gcloud.sh
-  ansible.builtin.shell: bash -o errexit -o pipefail /tmp/install-gcloud.sh --disable-prompts --install-dir=/
-
-- name: Remove install-gcloud.sh
-  ansible.builtin.file:
-    path: /tmp/install-gcloud.sh
-    state: absent
-
-- name: Find all files in /google-cloud-sdk/bin/
-  ansible.builtin.find:
-    paths: /google-cloud-sdk/bin/
-  register: find
-
-- name: Create symlinks to /bin
-  become: true
-  ansible.builtin.file:
-    src: "{{ item.path }}"
-    path: /bin/{{ item.path | basename }}
-    state: link
-  with_items: "{{ find.files }}"
+- name: Install gcloud SDK
+  when: ansible_os_family == "RedHat"
+  block:
+    - name: Add gcloud repository info
+      ansible.builtin.shell: |
+        sudo tee -a /etc/yum.repos.d/google-cloud-sdk.repo << EOM
+        [google-cloud-cli]
+        name=Google Cloud CLI
+        baseurl=https://packages.cloud.google.com/yum/repos/cloud-sdk-el8-x86_64
+        enabled=1
+        gpgcheck=1
+        repo_gpgcheck=0
+        gpgkey=https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
+        EOM
+    - name: Install google-cloud-cli package
+      ansible.builtin.yum:
+        name: google-cloud-cli
+        state: present
 
 - name: Install cloud-init packages
   ansible.builtin.yum:

